.github/actions/wait-for-check-completion/src/index.js

```diff
@@ -0,0 +1,194 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+/**
+ * Wait for multiple checks to complete
+ * @param {Object} octokit - GitHub API client
+ * @param {string} owner - Repository owner
+ * @param {string} repo - Repository name
+ * @param {string} ref - Git reference (commit SHA)
+ * @param {string[]} checkNames - Array of check names to wait for
+ * @param {number} waitInterval - Wait interval in seconds
+ * @param {number} timeout - Maximum timeout in seconds
+ * @returns {Promise<Object>} Combined check results
+ */
+async function waitForChecks(octokit, owner, repo, ref, checkNames, waitInterval, timeout) {
+    const startTime = Date.now();
+    const timeoutMs = timeout * 1000;
+    const waitIntervalMs = waitInterval * 1000;
+
+    core.info(`Waiting for checks [${checkNames.join(', ')}] on ref "${ref}" to complete...`);
+    core.info(`Timeout: ${timeout}s, Check interval: ${waitInterval}s`);
+
+    const checkResults = {};
+    const pendingChecks = new Set(checkNames);
+
+    while ((Date.now() - startTime < timeoutMs) && (pendingChecks.size > 0)) {
+        try {
+            // Get all check runs for the specific commit
+            const { data: checkRuns } = await octokit.rest.checks.listForRef({
+                owner,
+                repo,
+                ref,
+                per_page: 100
+            });
+
+            core.info(`Found ${checkRuns.check_runs.length} total check runs`);
+
+            // Process each pending check
+            for (const checkName of Array.from(pendingChecks)) {
```

Comment:
You can iterate over set, not necessary to convert back to array
```suggestion
            for (const checkName of pendingChecks) {
```